### PR TITLE
[Accessibility] Fix improperly nested Chips

### LINKED ISSUE #6762: ⌨ Dismissible chips improperly nested
## ⌨ Accessibility Issue

Our dismissible chips appear in a `ul` but are nested improperly as `list > button > listitem`.

### WCAG 2.1 Criteria

[Certain ARIA roles must contain particular children.](https://dequeuniversity.com/rules/axe/4.7/aria-required-children?application=axeAPI)

## 🦋 Expected Behaviour

The chips should be nested so that the `listitem` is a direct child of the parent `list`.

## 📋 Steps to Reproduce

1. Open the "Chip multi" story
2. Confirm the items are `listitem`s wrapped by `button`s and not the inverse

## 📸 Screenshot

Screenshot 2023-05-31 092408

## 🙋‍♀️ Proposed Solution

> **Warning**
> This could be a bit tricky!...or not 🤷‍♀️ 

- [ ] Update dismissible chips to to be nested as `listitem > button`

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the issue was addressed.

- [ ] Multiple, dismissible chips have the following HTML structure `list > listitem > button`
- [ ] No differences in the look and feel

### PR DESCRIPTION
**Pull request by yonikid15**
🤖 Resolves #6762 

## 👋 Introduction

- Ensures multiple, dismissible chips have the following HTML structure `list > listitem > button`

## 🧪 Testing

1. Open storybook and view Chips stories. Ensure there are no violations.
2. Ensure chips still look and feel the same on throughout site: SkillPicker, CommunitySelection, etc.

### CODE REVIEW

**Review by esizer — CHANGES_REQUESTED** [review]
Really cool solution. Just added a, hopefully small, suggestion 👍

File: packages/ui/src/components/Chip/Chips.tsx [hunk 1]
```diff
@@ -1,17 +1,29 @@
...
+        {children.map((child) => (
+          <div role="listitem" key={child.key}>
+            {child}
+          </div>
+        ))}
+      </div>
+    ) : (
+      children
+    )}
   </div>
```
[the remark this answers is not in the document]
Comment (yonikid15) [reply packages/ui/src/components/Chip/Chips.tsx:27]: 😆 Ya I wasn't sure if that was intentionally there or not

File: packages/ui/src/components/Chip/Chips.tsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 }
 
 const Chips = ({ children }: ChipsProps) => (
-  <div
-    role="list"
+  <ul
     data-h2-display="base(flex)"
     data-h2-flex-wrap="base(wrap)"
     data-h2-gap="base(x.25, x.125)"
   >
```
Comment (esizer) [inline packages/ui/src/components/Chip/Chips.tsx:14]:
```suggestion
  <ul
    data-h2-display="base(flex)"
    data-h2-flex-wrap="base(wrap)"
    data-h2-gap="base(x.25, x.125)"
    data-h2-list-style="base(none)"
  >
```

**Review by esizer — CHANGES_REQUESTED** [review]
Looking good, just need to remove that bullet now 👍

File: packages/ui/src/components/Chip/Chips.tsx (hunk 2 above)
Comment (yonikid15) [reply packages/ui/src/components/Chip/Chips.tsx]: Nice catch!

**Review by esizer — APPROVED** [review]
Nice! 🆒

**Pull request merged**